### DISCUSSION TITLE
Common Steps to troubleshoot networking issues

### DISCUSSION THREAD

**Discussion by Karsten75** [KnuckleCracker/CW4-bug-tracker#1036] [General]
Most often, when the servers at knucklecracker.com cannot be reached by a single user, it's indicative of an issue specific to one instance of web connectivity. 

Recommended steps to resolve:

1 Use an Internet-based detection tool to confirm that the problem is not, in fact at the knucklecracker website. 
   Popular  tools are: 
  - https://www.isitdownrightnow.com/
  - https://downforeveryoneorjustme.com/
  - Many more exist on the Internet and a quick Google search should find a few others. 
     
 Assuming the above website checkers report that KnuckleCracker.com is in fact available,     
   
2. Reboot your computer, and all associated in-home network device (Routers, switches, etc.)
3. Check the routing from your computer to knucklecracker.com
       https://support.microsoft.com/en-us/topic/how-to-use-tracert-to-troubleshoot-tcp-ip-problems-in-windows-e643d72b-2f4f-cdd6-09a0-fd2989c7ca8e

4.   Check your connection speed to the Linode site that hosts knucklecracker.com (Dallas, USA)
       https://www.linode.com/speed-test/

5. Test individual map downloads: 
      Example map for CW4: http://knucklecracker.com/creeperworld4/queryMaps.php?query=mapid&id=1  (Change ID to any desired, existing number)
      Example for CW3:   https://knucklecracker.com/creeperworld3/queryMaps.php?query=mapid&id=1  (Change ID to any desired, existing number)

**Comment by MartinX3** [comment]
Since today https://knucklecracker.com/creeperworld3/queryMaps.php?query=mapid&id=9304 shows me "403 Permission denied" and "ERR_CONNECTION_REFUSED"

Also I can't play any custom maps anymore.

**Comment by knucracker** [comment]
The server is currently being upgraded.  It should all be back within a few
hours.

**Reply by MartinX3** [reply]
Thank you.
In the steam forum "75Karsten" told us, the problem is at our end and we should fix our network with this guide.
So maybe he isn't informed like your customers.

**Reply by grabz-dev** [reply]
No, he has not told you that.

Don't conflate the information given to you 12 hours ago with information from 1 hour ago. The server went down on maintenance roughly 1:30 hours before this comment of mine, and is a completely unrelated event to whatever problem may have been had many hours prior by you or whoever.

**Reply by Karsten75** [reply]
> Thank you. In the steam forum "75Karsten" told us, the problem is at our end and we should fix our network with this guide. So maybe he isn't informed like your customers.

You will notice that the very first step in this article is a method to determine if the problem is only affecting a single user or is in fact, site-wide. :)
